bin/latexml

```diff
@@ -63,19 +64,24 @@ my $source = $ARGV[0];
 
 #**********************************************************************
 # Set up the processing.
-print STDERR "$LaTeXML::IDENTITY\n" if $verbosity > -1;
-print STDERR "processing started " . localtime() . "\n" if $verbosity > -1;
-
 @paths = map { pathname_canonical($_) } @paths;
 if (my @baddirs = grep { !-d $_ } @paths) {
   warn "These path directories do not exist: " . join(', ', @baddirs) . "\n"; }
 
+if (!pathname_is_literaldata($source)) {
+  my ($dir, $name, $ext) = pathname_split($source);
+  if ($name) {
+    my $path = pathname_make(dir => pathname_cwd(), name => $name, type => 'lxlog');
```

Comment:
For latexmlc, the log stuff should have moved to `LaTeXML.pm` (or `Core`, or ...), rather than `latexmlc`. Ultimately, should be removed from `latexml`,`latexmlpost`, I think. The catch is that I don't quite find the layers/factoring in `LaTeXML.pm` to deal with when/where to open log (after config), vs reporting about problems w/config. Plus (I guess) there're config for executable & config per job.  So, will be good to see your take on things.

As far as the log file name, we're expecting folks to also run (pdf)latex, so there will already be a `.log` file and we'd be bad neighbors to overwrite it. So, unless we discover that the natural place for latexml's log is in a different directory, we'll need a different name.  (If you don't like `.lxlog`, you could always use `--log` :>)

Reply:
Well, I don't like `.lxlog` to a point that I feel it shouldn't exist really... If there's a worry that `$jobname.log` is taken by pdflatex, we could do `.tex.log` (similar to how we've done `.tex.xml`) or `$jobname_$executable.log` i.e. `$jobname_latexml.log`, `$jobname_latexmlpost.log`, etc. Using a `.log` ultimate extension makes the setup quite standard and predictable for outsiders (and for me :>).

Reply:
I suppose we could achieve that with `--log=none`, then.

Reply:
Standard and predictable for anyone that uses tex/latex/pdflatex is that it not overwrite `$jobname.log`. No "if".

Reply:
For your consideration, freshly sampling from the wider conversion ecosystem:
 - `htlatex` (and tex4ht generally), uses `$jobname.log` as the default logging target
 - `tralics` uses `$jobname.log` as the default logging target
 - `pandoc` has no default logging target, but it logs its subprocess calls (as it calls external executables to create PDFs) to e.g. `$jobname.log.pdflatex` and `$jobname.log.lualatex`, depending on the delegate.
 - `latex2html` has no default logging target, but uses `$$_images.log` for image conversion subprocesses
 - `plastex` has no default logging target unless requested with `--log`, which unconditionally redirects STDERR to `$jobname.log` without accepting any custom filename

The one thing I'll restate is that using the `.log` extension is definitely something that's easy to conform to. Whether colliding with the pdflatex log file is important or not I feel neutral on, perfectly happy with using a latexml-unique suffix for the filenames.

Reply:
And for yours: `bibtex` and `makeindex` create `.blg` and .`ilg`, respectively.  The two programs tex4ht and latex2html are creating those `.log` files exactly because they are actually running some version of TeX on certain files and get whatever it created; no choices involved. It seems that the pandoc & plastex folks realized that automatically overwriting the `.log` file was a bad idea, even if you don't, so they came up with alternatives. I'll let tralics make its own choices, but feel no obligation to copy them.

I don't quite understand the comment about "easy to conform to". It's just as easy to conform to the log file *not* being `.log'.

At any rate, this is kinda my least worry at this stage, but `$jobname.latexml.log` would be reasonable, if you preferred.

Reply:
Completely happy with `$jobname.latexml.log`, looks great! Best of all worlds.